### ISSUE TITLE
MonetDB crashes with a SIGSEGV due to a null pointer dereference when using prepared statements

### ISSUE THREAD

**Issue by gm-digitalk** [MonetDB/MonetDB#7537]
**Describe the bug**
MonetDB crashes with a SIGSEGV due to a null pointer dereference when using prepared statements with multiple connections and DDL operations.

**To Reproduce**
Repro SQL -- note that this assumes that the prepared statement id will start at 0 - adjust the exec statement below as required.

_Connection 1:_

 drop function if exists test_epoch_ms;
 drop view if exists v3;
 drop view if exists v4;
 drop procedure if exists test_pop();
 drop table if exists t1;
 drop table if exists t2;
 drop table if exists t3;
 drop table if exists t4;

 create function test_epoch_ms(ts timestamp) returns bigint external name mtime.epoch_ms;
 create table t1 (c1 integer, c2 timestamp);
 create table t2 (c1 integer, c2 timestamp);
 create table t3 (c1 integer, c2 timestamp);
 create table t4 (c1 integer, c2 timestamp);
 create view v3 as select * from t1 union all select * from t2 union all select * from t3;
 create view v4 as select * from t1 union all select * from t2 union all select * from t3 union all select * from t4;
 create procedure test_pop() begin declare i int; set i = 0; while i < 10000 do insert into t1 values (i, current_timestamp()); insert into t2 values (10000 + i, current_timestamp()); insert into t3 values (20000 + i, current_timestamp()); insert into t4 values (30000 + i, current_timestamp()); set i = i + 1; end while; end;
 call test_pop();
 drop procedure if exists test_pop();
 alter table t1 set read only;
 alter table t2 set read only;
 alter table t3 set read only;
 alter table t4 set read only;

 prepare with agg1(agg1_avg) as (select avg(v31.c1) from v3 as v31 join v3 v32 on 1) select avg((agg1.agg1_avg + test_epoch_ms(v4.c2))) as c2 from agg1 join v4 on ?;

 exec 0(1);

_While the above is running on Connection 2:_

 drop view if exists v4; drop table if exists t4;

_As soon as a prompt is available on Connection 1:_

 exec 0(1);

**Expected behavior**
Not crashing and failing the prepared statement query.

**Software versions**
 - MonetDB version number 11.45.13 and 11.49.9
 - OS and version: Debian 10 Buster, Debian 11 Bullseye
 - Installed from release package

**Additional context**
gdb of core (edited to add core monetdb version is 11.45.13):

```
Core was generated by `/usr/bin/mserver5 --dbpath=/data/dbfarm/######### --set=merovingian_uri mapi:mo'.
Program terminated with signal SIGSEGV, Segmentation fault.
#0  0x00007fb85a978e34 in find_sql_column (t=t@entry=0x0, cname=cname@entry=0x7fb78e5152d0 "b#################") at ./sql/storage/sql_catalog.c:246
246             node *n = ol_find_name(t->columns, cname);
[Current thread is 1 (Thread 0x7fb84d7ec700 (LWP 2474))]
>>> info threads
  Id   Target Id                         Frame
* 1    Thread 0x7fb84d7ec700 (LWP 2474)  0x00007fb85a978e34 in find_sql_column (t=t@entry=0x0, cname=cname@entry=0x7fb78e5152d0 "b#################")") at ./sql/storage/sql_catalog.c:246
  2    Thread 0x7fb84bfe0700 (LWP 2486)  0x00007fb85a978e34 in find_sql_column (t=t@entry=0x0, cname=cname@entry=0x7fb78e5152a0 "asig_netgroup") at ./sql/storage/sql_catalog.c:246
  3    Thread 0x7fb85b27e980 (LWP 2309)  0x00007fb85e477bf0 in __GI___nanosleep (requested_time=requested_time@entry=0x7ffc363684b0, remaining=remaining@entry=0x0) at ../sysdeps/unix/sysv/linux/nanosleep.c:28
  4    Thread 0x7fb84bddf700 (LWP 2487)  0x00007fb85a978e34 in find_sql_column (t=t@entry=0x0, cname=cname@entry=0x7fb78e514ff0 "asig_invite") at ./sql/storage/sql_catalog.c:246
  5    Thread 0x7fb84ddef700 (LWP 2471)  __GI___fxstat (vers=vers@entry=1, fd=fd@entry=13, buf=buf@entry=0x7fb84ddee4e0) at ../sysdeps/unix/sysv/linux/wordsize-64/fxstat.c:35
  6    Thread 0x7fb84e3f2700 (LWP 2468)  __GI___fxstat (vers=vers@entry=1, fd=fd@entry=11, buf=buf@entry=0x7fb84e3f14e0) at ../sysdeps/unix/sysv/linux/wordsize-64/fxstat.c:35
  7    Thread 0x7fb84e1f1700 (LWP 2469)  __lll_lock_wait () at ../sysdeps/unix/sysv/linux/x86_64/lowlevellock.S:103
  8    Thread 0x7fb84d1e9700 (LWP 2477)  __libc_disable_asynccancel () at ../sysdeps/unix/sysv/linux/x86_64/cancellation.S:85
  9    Thread 0x7fb84dff0700 (LWP 2470)  0x00007fb85e477d0e in __libc_open64 (file=file@entry=0x7fb81000c3a0 "/data/dbfarm/EquinoxVP/bat/52/5267.tail2", oflag=oflag@entry=524290) at ../sysdeps/unix/sysv/linux/open64.c:48
  10   Thread 0x7fb84cfe8700 (LWP 2478)  0x00007fb85e58531c in q_dequeue (q=0x55d8513ce9c0, cntxt=cntxt@entry=0x0) at ./monetdb5/mal/mal_dataflow.c:261
  11   Thread 0x7fb84cde7700 (LWP 2479)  0x00007fb85a979eb1 in find_name (os=0x55d850fc7330, name=name@entry=0x7fb78e64c900 "c##########") at ./sql/storage/objectset.c:703
  12   Thread 0x7fb84cbe6700 (LWP 2480)  0x00007fb85e572c07 in initialize_tl_client_key () at ./monetdb5/mal/mal.c:69
  13   Thread 0x7fb84d5eb700 (LWP 2475)  futex_abstimed_wait_cancelable (private=0, abstime=0x0, expected=0, futex_word=0x7fb85e7b7d78 <workers+105592>) at ../sysdeps/unix/sysv/linux/futex-internal.h:205
  14   Thread 0x7fb84c5e3700 (LWP 2483)  0x00007fb85e4775d7 in __close (fd=fd@entry=15) at ../sysdeps/unix/sysv/linux/close.c:27
  15   Thread 0x7fb84c3e2700 (LWP 2484)  0x00007fb85e3946f9 in __GI___poll (fds=fds@entry=0x7fb84c3e1780, nfds=nfds@entry=1, timeout=timeout@entry=0) at ../sysdeps/unix/sysv/linux/poll.c:29
  16   Thread 0x7fb84bbde700 (LWP 2488)  __lll_lock_wait () at ../sysdeps/unix/sysv/linux/x86_64/lowlevellock.S:103
  17   Thread 0x7fb84c7e4700 (LWP 2482)  __lll_lock_wait () at ../sysdeps/unix/sysv/linux/x86_64/lowlevellock.S:103
  18   Thread 0x7fb84c1e1700 (LWP 2485)  rename (old=old@entry=0x7fb7f8000c50 "/data/dbfarm/EquinoxVP/bat/57/5740.tail1.new", new=new@entry=0x7fb7f8001300 "/data/dbfarm/EquinoxVP/bat/57/5740.tail1") at ../sysdeps/unix/sysv/linux/rename.c:29
  19   Thread 0x7fb84b3da700 (LWP 2492)  __lll_lock_wait () at ../sysdeps/unix/sysv/linux/x86_64/lowlevellock.S:103
  20   Thread 0x7fb84afd8700 (LWP 2494)  __lll_lock_wait () at ../sysdeps/unix/sysv/linux/x86_64/lowlevellock.S:103
  21   Thread 0x7fb84b7dc700 (LWP 2490)  __lll_lock_wait () at ../sysdeps/unix/sysv/linux/x86_64/lowlevellock.S:103
  22   Thread 0x7fb84a9d5700 (LWP 2497)  __lll_lock_wait () at ../sysdeps/unix/sysv/linux/x86_64/lowlevellock.S:103
  23   Thread 0x7fb848fc9700 (LWP 72690) 0x00007fb85e399cd3 in __GI___mmap64 (offset=0, fd=12, flags=1, prot=3, len=31064064, addr=0x0) at ../sysdeps/unix/sysv/linux/mmap64.c:59
  24   Thread 0x7fb84a3d2700 (LWP 2532)  0x00007fb85e3946f9 in __GI___poll (fds=fds@entry=0x7fb84a3d1c10, nfds=nfds@entry=1, timeout=100) at ../sysdeps/unix/sysv/linux/poll.c:29
  25   Thread 0x7fb84c9e5700 (LWP 2481)  __lll_lock_wait () at ../sysdeps/unix/sysv/linux/x86_64/lowlevellock.S:103
  26   Thread 0x7fb84d3ea700 (LWP 2476)  0x00007fb85e2f893a in buffered_vfprintf (s=s@entry=0x7fb85e461680 <_IO_2_1_stderr_>, format=format@entry=0x7fb85ee3cafb "#%s%s%s: %s: %s: %s%s%s\n", args=args@entry=0x7fb84d3e90c0) at vfprintf.c:2301
  27   Thread 0x7fb84b9dd700 (LWP 2489)  0x00007fb85e399cd3 in __GI___mmap64 (offset=0, fd=7, flags=1, prot=3, len=3932160, addr=0x0) at ../sysdeps/unix/sysv/linux/mmap64.c:59
  28   Thread 0x7fb85adf7700 (LWP 2342)  0x00007fb85e477bf0 in __GI___nanosleep (requested_time=requested_time@entry=0x7fb85adf6bd0, remaining=remaining@entry=0x0) at ../sysdeps/unix/sysv/linux/nanosleep.c:28
  29   Thread 0x7fb84a5d3700 (LWP 2531)  0x00007fb85e477bf0 in __GI___nanosleep (requested_time=requested_time@entry=0x7fb84a5d2bf0, remaining=remaining@entry=0x0) at ../sysdeps/unix/sysv/linux/nanosleep.c:28
  30   Thread 0x7fb84abd6700 (LWP 2496)  __lll_lock_wait () at ../sysdeps/unix/sysv/linux/x86_64/lowlevellock.S:103
  31   Thread 0x7fb84dbee700 (LWP 2472)  0x00007fb85a978e34 in find_sql_column (t=t@entry=0x0, cname=cname@entry=0x7fb78e5157f0 "sig_connected") at ./sql/storage/sql_catalog.c:246
  32   Thread 0x7fb81d5f9700 (LWP 40659) futex_abstimed_wait_cancelable (private=0, abstime=0x0, expected=0, futex_word=0x7fb78e6fa520) at ../sysdeps/unix/sysv/linux/futex-internal.h:205
  33   Thread 0x7fb849fd0700 (LWP 2617)  0x00007fb85e3946f9 in __GI___poll (fds=fds@entry=0x7fb849fcd980, nfds=nfds@entry=1, timeout=50) at ../sysdeps/unix/sysv/linux/poll.c:29
  34   Thread 0x7fb84add7700 (LWP 2495)  __lll_lock_wait () at ../sysdeps/unix/sysv/linux/x86_64/lowlevellock.S:103
  35   Thread 0x7fb84a7d4700 (LWP 2498)  __lll_lock_wait () at ../sysdeps/unix/sysv/linux/x86_64/lowlevellock.S:103
  36   Thread 0x7fb84a1d1700 (LWP 2533)  0x00007fb85e477bf0 in __GI___nanosleep (requested_time=requested_time@entry=0x7fb84a1d0c50, remaining=remaining@entry=0x0) at ../sysdeps/unix/sysv/linux/nanosleep.c:28
  37   Thread 0x7fb84b1d9700 (LWP 2493)  0x00007fb85a978e34 in find_sql_column (t=t@entry=0x0, cname=cname@entry=0x7fb78e5154e0 "bsig_netgroup") at ./sql/storage/sql_catalog.c:246
  38   Thread 0x7fb84b5db700 (LWP 2491)  0x00007fb85a978e34 in find_sql_column (t=t@entry=0x0, cname=cname@entry=0x7fb78e515270 "asig_carrier_group") at ./sql/storage/sql_catalog.c:246
  39   Thread 0x7fb84e5f3700 (LWP 2467)  0x00007fb85a978e34 in find_sql_column (t=t@entry=0x0, cname=cname@entry=0x7fb78e5158c0 "sig_end") at ./sql/storage/sql_catalog.c:246
  40   Thread 0x7fb84d9ed700 (LWP 2473)  0x00007fb85a978e34 in find_sql_column (t=t@entry=0x0, cname=cname@entry=0x7fb78e515540 "lcr_zone") at ./sql/storage/sql_catalog.c:246
>>> backtrace
#0  0x00007fb85a978e34 in find_sql_column (t=t@entry=0x0, cname=cname@entry=0x7fb78e5152d0 "b#################")") at ./sql/storage/sql_catalog.c:246
#1  0x00007fb85a9bcab5 in mvc_bind_column (m=<optimized out>, t=0x0, cname=cname@entry=0x7fb78e5152d0 "b#################")") at ./sql/server/sql_mvc.c:953
#2  0x00007fb85a8a3caf in mvc_bind_wrap (cntxt=<optimized out>, mb=0x7fb78e271c00, stk=0x7fb78ef63d10, pci=0x7fb78d0be530) at ./sql/backends/monet5/sql.c:1255
#3  0x00007fb85e582df0 in runMALsequence (cntxt=<optimized out>, mb=<optimized out>, startpc=<optimized out>, stoppc=63, stk=<optimized out>, env=env@entry=0x0, pcicaller=0x0) at ./monetdb5/mal/mal_interpreter.c:681
#4  0x00007fb85e5856f0 in DFLOWworker (T=0x7fb85e7b7dc0 <workers+105664>) at ./monetdb5/mal/mal_dataflow.c:379
#5  DFLOWworker (T=0x7fb85e7b7dc0 <workers+105664>) at ./monetdb5/mal/mal_dataflow.c:298
#6  0x00007fb85ec45a7b in THRstarter (a=0x55d8513c2660) at ./gdk/gdk_utils.c:1658
#7  0x00007fb85ec8a499 in thread_starter (arg=0x55d8511941b0) at ./gdk/gdk_system.c:862
#8  0x00007fb85e46dfa3 in start_thread (arg=<optimized out>) at pthread_create.c:486
#9  0x00007fb85e39f06f in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95

```

**Comment by gm-digitalk** [comment]
Edited issue to clarify monetdb version of core dump and correct a typo for the monetdb version from 11.43.13 to 11.45.13.

**Comment by lrpereira** [comment]
Thanks for the report.
Fixed in Aug2024 branch.

**Comment by lrpereira** [comment]
Grafted this fix into Dec2023.
